### PR TITLE
add and remove finalizer host definition

### PR DESCRIPTION
**Pull request by matancarmeli7**
Signed-off-by: matancarmeli7 <matan.carmeli7@gmail.com>

### CODE REVIEW

File: controllers/servers/host_definer/kubernetes_manager/manager.py [hunk 1]
```diff
@@ -169,27 +169,21 @@ def _is_host_definition_matches(self, host_definition_info, node_name, secret_na
...
         try:
             k8s_host_definition = self.host_definitions_api.create(body=host_definition_manifest)
+            self._add_finalizer_to_host_definition(k8s_host_definition.metadata.name)
             return self._generate_host_definition_info(k8s_host_definition)
         except ApiException as ex:
             if ex != 404:
                 logger.error(messages.FAILED_TO_CREATE_HOST_DEFINITION.format(
                     host_definition_manifest[settings.METADATA][settings.NAME], ex.body))
 
+    def _add_finalizer_to_host_definition(self, host_definition_name):
```
Comment (ArbelNathan) [inline controllers/servers/host_definer/kubernetes_manager/manager.py:182]: I think _add_finalizer is enough (as the function gets `host_definition_name`).
same for update

Comment (matancarmeli7) [reply controllers/servers/host_definer/kubernetes_manager/manager.py]: done

**Review by ArbelNathan — DISMISSED** [review]
LGTM

**Review by ELENAGER — APPROVED** [review]
LGTM

**Review by ArbelNathan — APPROVED** [review]

File: controllers/servers/host_definer/settings.py [hunk 2]
```diff
@@ -47,3 +47,4 @@
 SUCCESSFUL_MESSAGE_TYPE = 'Successful'
 NORMAL_EVENT_TYPE = 'Normal'
 WARNING_EVENT_TYPE = 'Warning'
+FINALIZERS = 'finalizers'
```
Comment (ArbelNathan) [inline controllers/servers/host_definer/settings.py:50]: @matancarmeli7 could we add some context to `FINALIZERS`?

Comment (matancarmeli7) [reply controllers/servers/host_definer/settings.py:50]: `FINALIZERS_KEY` will be good?

**Pull request merged**